### PR TITLE
PWGHF: update SigmaC creator and track-to-coll. associator.

### PR DESCRIPTION
**Pull request by mfaggin**
1. insert a new mask for Filters with only ITS quality cuts (useful for SigmaC soft pions)
 2. configurable for track-selections in track-to-coll. associator (good also for other developments)
 3. use re-associated tracks for SigmaC soft pions and add recalculation of dcaXY, dcaZ for reassigned tracks

### CODE REVIEW

**Comment by mfaggin** [comment]
@fcatalan92 @lvermunt FYK as well

**Review by lvermunt — DISMISSED** [review]

**Review by lvermunt — APPROVED** [review]

**Review by vkucera — CHANGES_REQUESTED** [review]

File: PWGHF/TableProducer/candidateCreatorSigmac0plusplus.cxx [hunk 1]
```diff
@@ -80,94 +113,140 @@ struct HfCandidateCreatorSigmac0plusplus {
...
+    ccdb->setCaching(true);
+    ccdb->setLocalObjectValidityChecking();
+    lut = o2::base::MatLayerCylSet::rectifyPtrFromFile(ccdb->get<o2::base::MatLayerCylSet>(ccdbPathLut));
+    runNumber = 0;
   }
 
-  /// @brief process function for Σc0,++ → Λc+(→pK-π+) π- candidate reconstruction
+  Preslice<aod::TrackAssoc> trackIndicesPerCollision = aod::track_association::collisionId;
+  // Preslice<CandidatesLc> hf3ProngPerCollision = aod::track_association::collisionId;
+  Preslice<CandidatesLc> hf3ProngPerCollision = aod::hf_cand::collisionId;
```
[the remark this answers is not in the document]
Comment (mfaggin) [reply PWGHF/TableProducer/candidateCreatorSigmac0plusplus.cxx:127]: Done

**Comment by mfaggin** [comment]
thanks @vkucera , I implemented your request

**Comment by vkucera** [comment]
> thanks @vkucera , I implemented your request

Thanks @mfaggin . There are still some related compilation issues.

**Comment by mfaggin** [comment]
> > thanks @vkucera , I implemented your request
> 
> Thanks @mfaggin . There are still some related compilation issues.

sorry @vkucera you are right. Everything should be ok now

**Review by vkucera — DISMISSED** [review]

**Comment by mfaggin** [comment]
thanks Vit for the approval. @fcolamar @ginnocen  @jgrosseo @ddobrigk @iarsene can you please have a look? Thank you

File: Common/TableProducer/trackToCollisionAssociator.cxx [hunk 2]
```diff
@@ -35,12 +35,14 @@ struct TrackToCollisionAssociation {
   Configurable<float> nSigmaForTimeCompat{"nSigmaForTimeCompat", 4.f, "number of sigmas for time compatibility"};
   Configurable<float> timeMargin{"timeMargin", 0.f, "time margin in ns added to uncertainty because of uncalibrated TPC"};
   Configurable<bool> applyMinimalTrackSelForRun2{"applyMinimalTrackSelForRun2", false, "flag to apply minimal track selection for Run 2 in case of standard association"};
```
Comment (jgrosseo) [inline Common/TableProducer/trackToCollisionAssociator.cxx:37]: Would it be worth to include this also in the setTrackSelection flag?

**Comment by mfaggin** [comment]
Hi @jgrosseo thanks for the suggestion. I've just implemented it in the last commit. I decided to enable the Run2 cuts when `setTrackSelections == -1`, so that other positive integers are available for other Run3 cut sets. Let me know what you think.

**Review by jgrosseo — APPROVED** [review]
Thanks!

**Review by vkucera — APPROVED** [review]

**Pull request merged**